File: queue_.py
class Queue():
    def __init__(self, max_queue_size=False) -> None:
        self.queue = []
        self.start = 0
        self.end = 0
        if (max_queue_size):
            self.max_queue_size = max_queue_size

    def get_size(self):
        return len(self.queue)

    def insert(self, element):
        if (self.get_size() + 1 > self.max_queue_size):
            return False
        self.queue.append(element)
        return True

    def delete(self):
        try:
            return self.queue.pop(0)
        except IndexError:
            return False

    def clear(self):
        if self.get_size() == 0:
            return False
        while self.get_size() > 0:
            self.delete()
        return True

    def print_queue(self):
        print(self.queue)


# try:
#     MAX_STACK_SIZE = int(input("Enter maximum stack size [default 5]: "))
# except ValueError:
#     MAX_STACK_SIZE = 5
MAX_STACK_SIZE = 5

queue = Queue(MAX_STACK_SIZE)


while (True):
    queue.print_queue()  # for easy debugging

    print("\nAvailable queue operations: \n\
        1 Insert \n\
        2 Delete \n\
        3 Clear \n\
        4 Get Size \n\
        5 Exit \n\
        ")
    try:
        op = int(input("# Enter option: "))
    except ValueError:
        print("Invalid option")
        continue

    if (op == 1):
        temp = input("# Enter element to insert: ")
        if len(temp) == 0:
            print("# input can't be empty")
            continue
        output = queue.insert(temp)
        if (output):
            print("# {} inserted successfully".format(output))
        else:
            print("# QUEUE OVERFLOW !!!")

    elif (op == 2):
        temp = queue.delete()
        if (temp):
            print("# {} was deleted".format(temp))
        else:
            print("# ERROR: queue IS EMPTY")

    elif (op == 3):
        output = queue.clear()
        if (output):
            print("# queue has been cleared")
        else:
            print("# ERROR: queue is already empty")

    elif (op == 4):
        print("# queue size: {}".format(queue.get_size()))

    elif (op == 5):
        queue.clear()
        print("# queue has been cleared")
        print("# Exiting")
        break

File: queue/queue.c
#include <stdio.h>

// for demonstration purpose
#define MAX_SIZE 5

int insert(int queue[], int *rear, int item);
int delete(int queue[], int *front, int rear);
int getSize(int front, int rear);
int isEmpty(int front, int rear);
int isFull(int front, int rear);
int resetQueue(int *front, int *rear);
void print_queue(int queue[], int front, int rear);

void print_queue(int queue[], int front, int rear)
{
    printf("front: %d rear: %d\n", front, rear);

    for (int i = front + 1; i <= rear; i++)
        printf("%d ", queue[i]);
    printf("\n");
}

void main()
{
    int queue[MAX_SIZE];
    int front, rear;
    front = rear = -1;

    printf("\nInteger Queue Demonstration\n");

    int option, temp, output;
    while (option != 7)
    {
        print_queue(queue, front, rear); // for debugging
        printf("\n");

        printf("Available queue operations: \n\
        1 Insert \n\
        2 Delete \n\
        3 Get Size \n\
        4 isEmpty \n\
        5 isFull \n\
        6 Reset \n\
        7 Exit \n");

        printf("# Enter option: ");
        scanf("%d", &option);
        printf("\n");

        switch (option)
        {
        case 1:
            printf("# Enter element to insert: ");
            scanf("%d", &temp);

            output = insert(queue, &rear, temp);
            if (output)
                printf("\n# %d inserted successfully\n", temp);
            else
                printf("\n# ERROR: QUEUE OVERFLOW !!!\n");
            break;

        case 2:
            temp = delete (queue, &front, rear);
            switch (temp)
            {
            case -99999:
                printf("# ERROR: QUEUE UNDERFLOW\n");
                break;
            default:
                printf("# deleted %d\n", temp);
            }
            break;

        case 3:
            printf("# queue size: %d\n", getSize(front, rear));
            break;

        case 4:
            if (isEmpty(front, rear))
                printf("# queue is empty\n");
            else
                printf("# queue is NOT empty\n");
            break;

        case 5:
            if (isFull(front, rear))
                printf("# queue is full\n");
            else
                printf("# queue is NOT full\n");
            break;

        case 6:
            output = resetQueue(&front, &rear);
            if (output)
                printf("# queue has been reset\n");
            else
                printf("# ERROR: queue is already empty\n");
            break;

        case 7:
            resetQueue(&front, &rear);
            printf("# queue has been cleared\n");
            printf("# Exiting\n\n");
            break;

        default:
            printf("# INVALID OPTION\n");
        }
    }
}

int insert(int queue[], int *rear, int item)
{
    if (*rear == MAX_SIZE - 1) // queue overflow
        return 0;
    queue[++*rear] = item;
    return 1;
}

int delete(int queue[], int *front, int rear)
{
    if (*front == rear) // queue underflow
        return -99999;
    return queue[++*front];
}

int getSize(int front, int rear)
{
    return rear - front;
}

int isEmpty(int front, int rear)
{
    if (getSize(front, rear) == 0)
        return 1;
    return 0;
}

int isFull(int front, int rear)
{
    if (getSize(front, rear) == MAX_SIZE || front == MAX_SIZE - 1)
        return 1;
    return 0;
}

int resetQueue(int *front, int *rear)
{
    if (*rear == -1)
        return 0;
    *front = *rear = -1;

    return 1;
}
